src/libraries/System.Linq/src/System/Linq/Range.cs

```diff
@@ -24,35 +24,63 @@ public static IEnumerable<int> Range(int start, int count)
                 return [];
             }
 
-            return new RangeIterator(start, count);
+            return new RangeIterator<int>(start, count);
+        }
+
+        public static IEnumerable<T> Range<T>(T start, int count) where T : IBinaryInteger<T>
+        {
+            if (count < 0)
```

Comment:
Curious why this overload is using somewhat different validation flow compared to the existing method? Is it anticipating that `T.CreateTruncating` might be expensive?